### PR TITLE
Nicolas.laurenti/calculator make operations

### PR DESCRIPTION
**Pull request by jnicolaslaurenti**
the calculator, now make operations and when press the clean button, delete a value but when is long press, delete all the operation

Pixel 3A:
Calculator_Pixel

Nexus S:

Calculator_Nexus_S

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -75,7 +75,6 @@ fastlane/report.xml
 fastlane/Preview.html
 fastlane/screenshots
 fastlane/test_output
-fastlane/readme.md
```
Comment (SartiMau) [inline .gitignore]: why you remove the fastlane readme? this is not the readme that you have to use

File: ReadMe.md.txt [hunk 2]
```diff
@@ -0,0 +1,6 @@
+## 21/12/2020
+	*Calculator UI
```
Comment (SartiMau) [inline ReadMe.md.txt]: the .txt at the file is wrong. The extension should be Readme.md

File: app/src/main/java/com/onboarding/calculator/mvp/view/CalculatorView.java [hunk 3]
```diff
@@ -19,4 +23,18 @@ public CalculatorView(Activity activity, ActivityMainBinding binding) {
...
+        binding.textViewResult.setText(R.string.calculator_error);
+    }
+
+    @Override
+    public void showDeleteAllMessage(){
+        Toast.makeText(getContext(),MESSAGE_DELETE_ALL,Toast.LENGTH_LONG).show();
+    }
+    @Override
+    public void showDeleteMessage(){
+        Toast.makeText(getContext(),MESSAGE_DELETE_VALUE,Toast.LENGTH_LONG).show();
```
Comment (mauroLonghi) [inline app/src/main/java/com/onboarding/calculator/mvp/view/CalculatorView.java]: showDeleteAllMessage() and showDeleteMessage() are the same...maube you can use a common method with a parameter, to sent the text message toshow

File: app/src/main/java/com/onboarding/calculator/mvp/model/CalculatorModel.java [hunk 4]
```diff
@@ -70,6 +72,57 @@ public void setValues(String value) {
...
     }
 
+    public void delete() {
+        if (!operator.isEmpty() && secondOperand.isEmpty()) {
+            operator = EMPTY_STRING;
+        } else
+            switch (switchOp) {
+                case FIRST_OPERAND: {
+                    if (firstOperand.length() > ZERO_INT) {
+                        firstOperand = firstOperand.substring(0, firstOperand.length() - 1);
```
Comment (mauroLonghi) [inline app/src/main/java/com/onboarding/calculator/mvp/model/CalculatorModel.java]: remove harcodes please, here and in the next "case"

File: app/src/main/java/com/onboarding/calculator/mvp/view/CalculatorView.java [hunk 5]
```diff
@@ -19,4 +23,18 @@ public CalculatorView(Activity activity, ActivityMainBinding binding) {
...
     }
+
+    @Override
+    public void showError() {
+        binding.textViewResult.setText(R.string.calculator_error);
+    }
+
+    @Override
+    public void showDeleteAllMessage(){
+        Toast.makeText(getContext(),MESSAGE_DELETE_ALL,Toast.LENGTH_LONG).show();
```
Comment (SartiMau) [inline app/src/main/java/com/onboarding/calculator/mvp/view/CalculatorView.java]: add != null getContext validation

File: app/src/main/java/com/onboarding/calculator/util/ConstantsUtils.java [hunk 6]
```diff
@@ -7,7 +7,12 @@
     public static final String MUL = "*";
     public static final String DIV = "/";
     public static final String CLEAN = "C";
-    public static final String EQUAL = "=";
+    public static final String MESSAGE_DELETE_ALL = "Operation deleted";
```
Comment (SartiMau) [inline app/src/main/java/com/onboarding/calculator/util/ConstantsUtils.java]: put this message in the string resource

File: app/src/main/java/com/onboarding/calculator/util/ConstantsUtils.java [hunk 7]
```diff
@@ -7,7 +7,12 @@
...
+    public static final String MESSAGE_DELETE_VALUE = "Value deleted";
```
Comment (SartiMau) [inline app/src/main/java/com/onboarding/calculator/util/ConstantsUtils.java]: same

**Review by SartiMau — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/onboarding/calculator/MainActivity.java [hunk 8]
```diff
@@ -1,9 +1,9 @@
...
 
-import androidx.appcompat.app.AppCompatActivity;
-
 import android.os.Bundle;
 
+import androidx.appcompat.app.AppCompatActivity;
+
 import com.onboarding.calculator.databinding.ActivityMainBinding;
 import com.onboarding.calculator.mvp.contract.CalculatorContract;
 import com.onboarding.calculator.mvp.model.CalculatorModel;
```
Comment (nicolbidone) [inline app/src/main/java/com/onboarding/calculator/MainActivity.java:9]: you can use Ctrl + Alt + O command to optimize the imports

File: app/src/main/java/com/onboarding/calculator/mvp/model/CalculatorModel.java [hunk 9]
```diff
@@ -70,6 +72,57 @@ public void setValues(String value) {
...
+                    }
+                }
+                case SECOND_OPERAND: {
+                    if (secondOperand.length() > ZERO_INT) {
+                        secondOperand = secondOperand.substring(0, secondOperand.length() - 1);
+                    }
+                }
+            }
```
Comment (nicolbidone) [inline app/src/main/java/com/onboarding/calculator/mvp/model/CalculatorModel.java:93]: do you forget the break?

File: app/src/main/java/com/onboarding/calculator/mvp/model/CalculatorModel.java [hunk 10]
```diff
@@ -70,6 +72,57 @@ public void setValues(String value) {
...
+        } else
+            switch (switchOp) {
+                case FIRST_OPERAND: {
+                    if (firstOperand.length() > ZERO_INT) {
+                        firstOperand = firstOperand.substring(0, firstOperand.length() - 1);
+                    }
+                }
+                case SECOND_OPERAND: {
+                    if (secondOperand.length() > ZERO_INT) {
+                        secondOperand = secondOperand.substring(0, secondOperand.length() - 1);
```
Comment (nicolbidone) [inline app/src/main/java/com/onboarding/calculator/mvp/model/CalculatorModel.java]: please avoid magic numbers

File: app/src/main/java/com/onboarding/calculator/mvp/view/CalculatorView.java (hunk 3 above)
Comment (jnicolaslaurenti) [reply app/src/main/java/com/onboarding/calculator/mvp/view/CalculatorView.java]: Solved!

File: app/src/main/java/com/onboarding/calculator/mvp/model/CalculatorModel.java (hunk 4 above)
Comment (jnicolaslaurenti) [reply app/src/main/java/com/onboarding/calculator/mvp/model/CalculatorModel.java]: Solved!

File: .gitignore (hunk 1 above)
Comment (jnicolaslaurenti) [reply .gitignore]: Solved!

File: ReadMe.md.txt (hunk 2 above)
Comment (jnicolaslaurenti) [reply ReadMe.md.txt]: Solved!

File: app/src/main/java/com/onboarding/calculator/mvp/model/CalculatorModel.java [hunk 11]
```diff
@@ -70,6 +72,57 @@ public void setValues(String value) {
         }
     }
 
+    public void delete() {
+        if (!operator.isEmpty() && secondOperand.isEmpty()) {
+            operator = EMPTY_STRING;
+        } else
+            switch (switchOp) {
+                case FIRST_OPERAND: {
+                    if (firstOperand.length() > ZERO_INT) {
```
Comment (nicolbidone) [inline app/src/main/java/com/onboarding/calculator/mvp/model/CalculatorModel.java]: please give ZERO_INT a more representative name

File: app/src/main/java/com/onboarding/calculator/mvp/model/CalculatorModel.java [hunk 12]
```diff
@@ -70,6 +72,57 @@ public void setValues(String value) {
...
+                    if (secondOperand.length() > ZERO_INT) {
+                        secondOperand = secondOperand.substring(0, secondOperand.length() - 1);
+                    }
+                }
+            }
+    }
+
+    @Override
+    public Double getResult() {
+        Double result = ZERO_DOUBLE;
```
Comment (nicolbidone) [inline app/src/main/java/com/onboarding/calculator/mvp/model/CalculatorModel.java]: please give ZERO_DOUBLE a more representative name

File: app/src/main/java/com/onboarding/calculator/mvp/view/CalculatorView.java (hunk 5 above)
Comment (jnicolaslaurenti) [reply app/src/main/java/com/onboarding/calculator/mvp/view/CalculatorView.java]: Solved!

File: app/src/main/java/com/onboarding/calculator/mvp/model/CalculatorModel.java [hunk 13]
```diff
@@ -70,6 +72,57 @@ public void setValues(String value) {
...
+            case DIV: {
+                try {
+                    if (!secondOperand.equals(ZERO_STRING)) {
+                        result = Double.parseDouble(firstOperand) / Double.parseDouble(secondOperand);
+                    } else {
+                        result = null;
+                    }
+                } catch (Exception exception) {
+                    result = null;
+                }
```
Comment (nicolbidone) [inline app/src/main/java/com/onboarding/calculator/mvp/model/CalculatorModel.java:117]: Do you need this try-catch?

File: app/src/main/java/com/onboarding/calculator/MainActivity.java (hunk 8 above)
Comment (jnicolaslaurenti) [reply app/src/main/java/com/onboarding/calculator/MainActivity.java:9]: Solved!

File: app/src/main/java/com/onboarding/calculator/mvp/model/CalculatorModel.java (hunk 10 above)
Comment (jnicolaslaurenti) [reply app/src/main/java/com/onboarding/calculator/mvp/model/CalculatorModel.java]: Solved!

File: app/src/main/java/com/onboarding/calculator/util/ConstantsUtils.java (hunk 6 above)
Comment (jnicolaslaurenti) [reply app/src/main/java/com/onboarding/calculator/util/ConstantsUtils.java]: Solved!

File: app/src/main/java/com/onboarding/calculator/util/ConstantsUtils.java (hunk 7 above)
Comment (jnicolaslaurenti) [reply app/src/main/java/com/onboarding/calculator/util/ConstantsUtils.java]: Solved!

File: app/src/main/java/com/onboarding/calculator/mvp/model/CalculatorModel.java (hunk 9 above)
Comment (jnicolaslaurenti) [reply app/src/main/java/com/onboarding/calculator/mvp/model/CalculatorModel.java:93]: Yes, I forgot the breaks. Solved!

File: app/src/main/java/com/onboarding/calculator/mvp/model/CalculatorModel.java (hunk 11 above)
Comment (jnicolaslaurenti) [reply app/src/main/java/com/onboarding/calculator/mvp/model/CalculatorModel.java]: Solved!

File: app/src/main/java/com/onboarding/calculator/mvp/model/CalculatorModel.java (hunk 12 above)
Comment (jnicolaslaurenti) [reply app/src/main/java/com/onboarding/calculator/mvp/model/CalculatorModel.java]: Solved!

File: app/src/main/java/com/onboarding/calculator/mvp/model/CalculatorModel.java (hunk 13 above)
Comment (jnicolaslaurenti) [reply app/src/main/java/com/onboarding/calculator/mvp/model/CalculatorModel.java:117]: Solved!

**Review by nicolbidone — APPROVED** [review]

**Review by BQuinteros — APPROVED** [review]

**Review by LuchoElissondoGlobant — APPROVED** [review]

File: app/src/main/java/com/onboarding/calculator/mvp/presenter/CalculatorPresenter.java [hunk 14]
```diff
@@ -20,4 +18,31 @@ public void onCalculatorButtonPressed(String buttonText) {
...
+        }
+
+    }
+
+    @Override
+    public boolean deleteAll() {
+        model.reset();
+        view.showValues(model.getLastModified());
+        view.showDeleteAllMessage();
+        return true;
```
Comment (SartiMau) [inline app/src/main/java/com/onboarding/calculator/mvp/presenter/CalculatorPresenter.java]: why do you need this boolean return? you are not using it on the activity

File: app/src/main/java/com/onboarding/calculator/mvp/presenter/CalculatorPresenter.java [hunk 15]
```diff
@@ -20,4 +18,31 @@ public void onCalculatorButtonPressed(String buttonText) {
...
+    }
+
+    @Override
+    public boolean delete() {
+        model.delete();
+        view.showValues(model.getLastModified());
+        view.showDeleteMessage();
+        return true;
```
Comment (SartiMau) [inline app/src/main/java/com/onboarding/calculator/mvp/presenter/CalculatorPresenter.java]: same here

**Review by SartiMau — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/onboarding/calculator/mvp/presenter/CalculatorPresenter.java (hunk 15 above)
Comment (jnicolaslaurenti) [reply app/src/main/java/com/onboarding/calculator/mvp/presenter/CalculatorPresenter.java]: Solved!

File: app/src/main/java/com/onboarding/calculator/mvp/presenter/CalculatorPresenter.java (hunk 14 above)
Comment (jnicolaslaurenti) [reply app/src/main/java/com/onboarding/calculator/mvp/presenter/CalculatorPresenter.java]: Solved!

**Review by SartiMau — APPROVED** [review]

**Review by mauroLonghi — APPROVED** [review]

File: app/src/main/java/com/onboarding/calculator/mvp/view/CalculatorView.java (hunk 3 above)
Comment (mauroLonghi) [reply app/src/main/java/com/onboarding/calculator/mvp/view/CalculatorView.java]: comment not fixed,

**Pull request merged**